### PR TITLE
RATIS-2035. Refactor streaming code for Read.

### PR DESCRIPTION
**Pull request by szetszwo**
See RATIS-2035

### CODE REVIEW

File: ratis-server-api/src/main/java/org/apache/ratis/server/DataStreamServerRpc.java [hunk 1]
```diff
@@ -20,18 +20,12 @@
...
  * A server interface handling incoming streams
  * Relays those streams to other servers after persisting
  */
-public interface DataStreamServerRpc extends RaftPeer.Add, Closeable {
-  /**
-   * start server
-   */
+public interface DataStreamServerRpc extends ServerRpc, RaftPeer.Add, Closeable {
+  @Override
   void start();
```
Comment (OneSizeFitsQuorum) [inline ratis-server-api/src/main/java/org/apache/ratis/server/DataStreamServerRpc.java:30]: Do we need to keep this function as ServerRpc already has one?

Comment (szetszwo) [reply ratis-server-api/src/main/java/org/apache/ratis/server/DataStreamServerRpc.java:30]: You are right that we can remove it, although this signature in this one is different -- it does not `throws IOException`.

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM~
The code becomes much clearer ~

**Review by adoroszlai — APPROVED** [review]

**Comment by adoroszlai** [comment]
Thanks @szetszwo for the patch, @OneSizeFitsQuorum for the review.

**Comment by szetszwo** [comment]
@adoroszlai , thanks a lot for reviewing and merging this!

**Pull request merged**